### PR TITLE
Update known issues for WPILib 2022.4.1

### PR DESCRIPTION
**Pull request by sciencewhiz**
Add known issue for synchronized deadlock

### CODE REVIEW

**Comment by PeterJohnson** [comment]
Now that 2022.4.1 is released we can move it to the "fixed" section.

**Comment by sciencewhiz** [comment]
Is the technical writeup ok?

File: source/docs/yearly-overview/known-issues.rst [hunk 1]
```diff
@@ -119,6 +112,41 @@ LabVIEW Autorun graphics say 2020
...
+------------------------
+
+Reentrant uses of synchronized may cause deadlock in Java
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Issue:** If multiple threads call a `synchronized` object, there may be a deadlock. This is caused by a bug in the version of JDK bundled with the 2022.1.1 to 2022.3.1 versions of WPILib.
+
+**Workaround:** There are two options for workarounds:
+
+#. Install the 2021 JDK. This is performed automatically in the WPILib 2022.4.1 and later.
```
Comment (PeterJohnson) [inline source/docs/yearly-overview/known-issues.rst:125]:
```suggestion
#. Install the 2021 JDK. This is performed automatically in WPILib 2022.4.1 and later.
```

**Comment by PeterJohnson** [comment]
In general looks good to me.

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**